moto/cloudformation/models.py

```diff
@@ -325,11 +327,27 @@ def stack_outputs(self):
     def exports(self):
         return self.output_map.exports
 
+    def add_custom_resource(self, custom_resource):
+        self.custom_resources[custom_resource.logical_id] = custom_resource
+
+    def get_custom_resource(self, custom_resource):
+        return self.custom_resources[custom_resource]
+
     def create_resources(self):
-        self.resource_map.create(self.template_dict)
+        self.status = "CREATE_IN_PROGRESS"
```

Comment:
This makes the Stack-creation async (-ish). It's marked as create-in-progress - unless all of the resources are created immediately, then we can set the status as 'create-complete'